### PR TITLE
bump TestHTTP1DoNotReuseRequestAfterTimeout timeout

### LINKED ISSUE #106569: [Flaky Test][sig-node] kubernetes-unit-test TestHTTP1DoNotReuseRequestAfterTimeout is being Flaky
### Which jobs are flaking?

periodic-kubernetes-unit-test-ppc64le

### Which tests are flaking?

1. vendor/k8s.io/client-go/rest.TestHTTP1DoNotReuseRequestAfterTimeout
2. vendor/k8s.io/client-go/rest.TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1
3. vendor/k8s.io/client-go/rest.TestHTTP1DoNotReuseRequestAfterTimeout/HTTP2

### Since when has it been flaking?

16th November 2021

### Testgrid link

https://k8s-testgrid.appspot.com/sig-node-ppc64le#unit-tests

### Reason for failure (if possible)

This test is added by change https://github.com/kubernetes/kubernetes/pull/104844

Seems like it started being flaky from the time added.

### Anything else we need to know?

Below is the trace from job:
```
 === FAIL: vendor/k8s.io/client-go/rest TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.13s)
    request_test.go:3016: Unexpected error: Get "https://127.0.0.1:34701/foo?timeout=100ms": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2021/11/20 12:06:03 http: TLS handshake error from 127.0.0.1:37890: read tcp 127.0.0.1:34701->127.0.0.1:37890: use of closed network connection
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.13s)
=== FAIL: vendor/k8s.io/client-go/rest TestHTTP1DoNotReuseRequestAfterTimeout (0.30s) 
```

### Relevant SIG(s)

/sig node

### PR DESCRIPTION
**Pull request by aojea**
the test TestHTTP1DoNotReuseRequestAfterTimeout has to wait for
request to time out to assert that subsequent requests does not
reuse the TCP connection.

It seems that current value of 100ms causes issues on some CI
environments and bumping the timeout seems to solve this flakiness,

We can bump the timeout value because is really low compared to real
scenarios and the bump still keeps it in the millisecond order.

/kind flake
Fixes #106569

#### Special notes for your reviewer:

golang CI suffer similar issues too
https://github.com/kubernetes/kubernetes/issues/106569#issuecomment-974704197

```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
/test pull-kubernetes-unit
```

k8s.io/kubernetes/pkg/controller/statefulset: TestStatefulSetControl/ScalesUp/Burst/Delete/StatefulSetAutoDeletePVCEnabled expand_less	0s
=== RUN   TestStatefulSetControl/ScalesUp/Burst/Delete/StatefulSetAutoDeletePVCEnabled
W1129 14:57:16.584354   52919 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W1129 14:57:16.584962   52919 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W1129 14:57:16.585175   52919 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W1129 14:57:16.585251   52919 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
    stateful_set_control_test.go:220: Failed to turn up StatefulSet : persistentvolumeclaim "datadir-foo-0" not found
    stateful_set_control_test.go:224: Failed to scale StatefulSet : persistentvolumeclaim "datadir-foo-0" not found
    stateful_set_control_test.go:235: Failed to set readyReplicas correctly
```

**Comment by liggitt** [comment]
> k8s.io/kubernetes/pkg/controller/statefulset: TestStatefulSetControl/ScalesUp/Burst/Delete/StatefulSetAutoDeletePVCEnabled expand_less	0s

#106615

File: staging/src/k8s.io/client-go/rest/request_test.go [hunk 1]
```diff
@@ -2995,7 +2995,7 @@ func TestHTTP1DoNotReuseRequestAfterTimeout(t *testing.T) {
 			config := &Config{
 				Host:      ts.URL,
 				Transport: utilnet.SetTransportDefaults(transport),
-				Timeout:   100 * time.Millisecond,
+				Timeout:   300 * time.Millisecond,
```
[the remark this answers is not in the document]
Comment (aojea) [reply staging/src/k8s.io/client-go/rest/request_test.go:2998]: The test does the following:
1. send req and get the origin IP:port
2. send req and hang until time out
3. send req and get the origin IP: port

IP:port on 1 and 3 has to be different

The test were failing timing out trying to reach the "working" handlers on 1. or 3.
```
/tmp/go-stress-20211129T115302-2252016254
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.29s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.10s)
        request_test.go:2974: TEST Connected from 127.0.0.1:57294 on /foo
        request_test.go:3016: Unexpected error: Get "https://127.0.0.1:36767/foo?timeout=100ms": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
FAIL
```

a 100 ms for the roundtrip can be very strict , I think that is legit to allow more time for the requests
https://github.com/kubernetes/kubernetes/issues/106569#issuecomment-981590944

[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/client-go/rest/request_test.go:2998]: I see. 300ms still seems potentially flaky in starved CI envs

[the remark this answers is not in the document]
Comment (aojea) [reply staging/src/k8s.io/client-go/rest/request_test.go:2998]: agree, we can go with 1 sec

[the remark this answers is not in the document]
Comment (aojea) [reply staging/src/k8s.io/client-go/rest/request_test.go:2998]: is what other tests in the same file have

**Comment by liggitt** [comment]
/lgtm
/approve
/milestone v1.23
for flaky test fixup (should coordinate with release managers to ensure this gets to release-1.23 branch after merge as well)

**Pull request merged**